src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -293,6 +296,102 @@ private static void ThrowFormatException(object value, string fromType, string t
             throw new FormatException($"Could not convert {fromType} ('{value}') to {toType}", e);
         }
 
+        internal static bool TryConvertType<T>(object result, out T obj)
+        {
+            obj = default(T);
+            if (result == null)
+                return true;
+
+            if (result is T)
+            {
+                obj = (T)result;
+            }
+            //just in case -> have better exception in this use-case
+            else if (typeof(T) == typeof(BlittableJsonReaderObject) && result.GetType() == typeof(BlittableJsonReaderArray))
+            {
+                return false;
+            }
+            //just in case -> have better exception in this use-case
+            else if (typeof(T) == typeof(BlittableJsonReaderArray) && result.GetType() == typeof(BlittableJsonReaderObject))
+            {
+                return false;
+            }
+            else
+            {
+                var type = Nullable.GetUnderlyingType(typeof(T)) ?? typeof(T);
+
+                try
+                {
+                    if (type.GetTypeInfo().IsEnum)
+                    {
+                        obj = (T)Enum.Parse(type, result.ToString());
+                    }
+                    else if (type == typeof(DateTime))
+                    {
+                        if (ChangeTypeToString(result, out string dateTimeString) == false)
```

Comment:
Probably, but wouldn't be clean anyways because right now the core is unaware or span. The spanification of the core should deal with that and treat span as first class citizen starting wit voron and sparrow

Reply:
We can already use span, we have already started to take dependency on it.
So we can technically get a `Span<byte>` and then use `Utf8Parser.TryParse` directly

Reply:
Not sure how much that would help, the common case is a string, not date time

Reply:
Yes, I have several cases where I am already working with spans already, but they are still a clutch. We need a definite solution. The generalized allocators are giving us spans for arbitrary memory with a decent API. That along with the ability to customize heavily the behavior is the big change there.

Reply:
For example, if every call with a pointer you need to write `new ReadOnlySpan<byte>(ptr)` the code gets unwildly very fast.